tests/MongoDB.Driver.Tests/PipelineStageDefinitionBuilderTests.cs

```diff
@@ -103,10 +107,229 @@ public void ChangeStream_should_return_the_expected_result_when_options_isNull()
+            var warehousesCollection = client.GetDatabase("test").GetCollection<BsonDocument>("warehouses");
+
+            var pipeline = new EmptyPipelineDefinition<BsonDocument>();
+
+            var result = PipelineStageDefinitionBuilder.Lookup<BsonDocument, BsonDocument, BsonDocument, IEnumerable<BsonDocument>, BsonDocument >
+                (
+                    warehousesCollection,
+                    null,
+                    pipeline,
+                    new StringFieldDefinition<BsonDocument, IEnumerable<BsonDocument>>("stockdata")
+                );
+
+            RenderStage(result).Document.Should().Be(@"
+                {
+                    '$lookup' :
+                    {
+                        'from' : 'warehouses',
+                        'pipeline' : [ ],
+                        'as' : 'stockdata'
+                    }
+                }");
+        }
+
+        public class Order
+        {
+            [BsonElement("stockdata")]
+            public IEnumerable<StockData> StockData { get; set; }
+        }
+
+        public class Warehouse
+        {
+            [BsonElement("stock_item")]
+            public string StockItem { get; set; }
+            [BsonElement("instock")]
+            public int Instock { get; set; }
+        }
+
+        public class StockData
+        {
+            [BsonElement("instock")]
+            public int Instock { get; set; }
+        }
+
+        [Fact]
+        public void Lookup_with_entity_generic_params_should_return_the_expected_result()
+        {
+            RequireServer.Check().Supports(Feature.AggregateLet);
+
+            var client = new MongoClient(CoreTestConfiguration.ConnectionString.ToString());
+            var warehousesCollection = client.GetDatabase("test").GetCollection<Warehouse>("warehouses");
+
+            var pipeline = new EmptyPipelineDefinition<Warehouse>()
+                .Match(new BsonDocument("$expr",
+                    new BsonDocument("$and", new BsonArray
+                    {
+                        new BsonDocument("$eq", new BsonArray { "$stock_item", "$$order_item" }),
+                        new BsonDocument("$gte", new BsonArray { "$instock", "$$order_qty" })
+                    })))
+                .Project<Warehouse, Warehouse, StockData>(Builders<Warehouse>
+                    .Projection
+                    .Exclude(warehouses => warehouses.StockItem));
+
+            var result = PipelineStageDefinitionBuilder.Lookup<BsonDocument, Warehouse, StockData, IEnumerable<StockData>, Order>
+                (
+                    warehousesCollection,
+                    new BsonDocument
+                    {
+                        { "order_item", "$item" },
+                        { "order_qty", "$ordered" }
+                    },
+                    pipeline,
+                    new ExpressionFieldDefinition<Order, IEnumerable<StockData>>(order => order.StockData)
```

Comment:
We should be able to just write

    order => order.StockData

if we have an additional overload of Lookup that takes an Expression.